src/argparse.c

```diff
@@ -133,59 +137,209 @@ static inline bool file_exists(const char *const path)
     return -1 != stat(path, &buf);
 }
 
-static inline void load_env(neu_cli_args_t *args, char **log_level_out,
-                            char **config_dir_out, char **plugin_dir_out)
+static inline int load_env(neu_cli_args_t *args, char **log_level_out,
+                           char **config_dir_out, char **plugin_dir_out)
 {
 
+    int   ret    = 0;
     char *daemon = getenv(NEU_ENV_DAEMON);
     if (daemon != NULL) {
         if (strcmp(daemon, "1") == 0) {
             args->daemonized = true;
+        } else {
+            args->daemonized = false;
         }
     }
 
     char *log = getenv(NEU_ENV_LOG);
     if (log != NULL) {
         if (strcmp(log, "1") == 0) {
             args->dev_log = true;
+        } else {
+            args->dev_log = false;
         }
     }
 
     char *log_level = getenv(NEU_ENV_LOG_LEVEL);
     if (log_level != NULL) {
+        if (*log_level_out != NULL) {
+            free(*log_level_out);
+        }
         *log_level_out = strdup(log_level);
     }
 
     char *restart = getenv(NEU_ENV_RESTART);
     if (restart != NULL) {
-        int ret = parse_restart_policy(restart, &args->restart);
-        if (ret < 0) {
+        int t = parse_restart_policy(restart, &args->restart);
+        if (t < 0) {
             printf("neuron NEU_ENV_RESTART setting error!\n");
+            ret = -1;
         }
     }
 
     char *disable_auth = getenv(NEU_ENV_DISABLE_AUTH);
     if (disable_auth != NULL) {
         if (strcmp(disable_auth, "1") == 0) {
             args->disable_auth = true;
+        } else {
+            args->disable_auth = false;
         }
     }
 
     char *config_dir = getenv(NEU_ENV_CONFIG_DIR);
     if (config_dir != NULL) {
+        if (*config_dir_out != NULL) {
+            free(*config_dir_out);
+        }
         *config_dir_out = strdup(config_dir);
     }
 
     char *plugin_dir = getenv(NEU_ENV_PLUGIN_DIR);
     if (plugin_dir != NULL) {
+        if (*plugin_dir_out != NULL) {
+            free(*plugin_dir_out);
+        }
         *plugin_dir_out = strdup(plugin_dir);
     }
+
+    return ret;
+}
+
+static inline int load_config_file(char *config_file, neu_cli_args_t *args,
+                                   char **log_level_out, char **config_dir_out,
+                                   char **plugin_dir_out)
+{
+    char buf[512] = { 0 };
+    int  fd       = -1;
+    fd            = open(config_file, O_RDONLY);
+    if (fd < 0) {
+        printf("cannot open %s reason: %s\n", config_file, strerror(errno));
+        return -1;
+    }
+    int size = read(fd, buf, sizeof(buf) - 1);
+    if (size <= 0) {
+        printf("cannot read %s reason: %s\n", config_file, strerror(errno));
+        close(fd);
+        return -1;
+    } else {
```

Comment:
you don't need to put the main logic in the else branch since the `return` statement ends the control flow